client.go

```diff
@@ -26,18 +27,33 @@ var (
 	DefaultDirectoryURL = "https://plc.directory"
 )
 
-func (c *Client) Resolve(ctx context.Context, did string) (*Doc, error) {
-	if !strings.HasPrefix(did, "did:plc:") {
-		return nil, fmt.Errorf("expected a did:plc, got: %s", did)
+// turn a non-200 HTTP response into a descriptive error, and log the body
+func processErrorResponse(resp *http.Response, msg string) error {
+	if resp.StatusCode == http.StatusNotFound {
+		return ErrDIDNotFound
+	}
+
+	var body_str string
+	body := new(strings.Builder)
+	_, err := io.Copy(body, resp.Body)
+	if err != nil {
+		body_str = "<failed to read response body>"
+	} else {
+		body_str = body.String()
 	}
+	log.Println(body_str) // TODO: configure logging better
```

Comment:
I'd do something like:

```
import ( "log/slog" )

[...]

_, err := io.Copy(body, resp.Body)
if err != nil {
    slog.Info("failed reading PLC directory response body", "status_code", resp.StatusCode)
} else {
    slog.Info("PLC directory request failed", "status_code", resp.StatusCode, "body", body.String())
}
```

Separately, go has a fairly strong camelCase variable name norm vs snake_case (w/r/t `body_str`). I still tend to use snake_case in JSON field names and structured logging.